### PR TITLE
Fix for non-generally-sized-but-generic subclasses of @objc classes

### PR DESCRIPTION
**Pull request by slavapestov**
This cherry-picks the fix from master into swift-2.2-branch.

### CODE REVIEW

**Comment by slavapestov** [comment]
@jckarter Please review for Swift 2.2

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test

File: test/IRGen/generic_classes.sil [hunk 1]
```diff
@@ -313,8 +313,8 @@ entry(%c : $RootGeneric<Int32>):
 // CHECK:   call void @swift_initClassMetadata_UniversalStrategy(%swift.type* {{%.*}}, i64 3, i64* {{%.*}}, i64* {{%.*}})
 // CHECK: }
 
-// CHECK: define private %swift.type* @create_generic_metadata_RootGenericFixedLayout(%swift.type_pattern*, i8**) {{.*}} {
-// CHECK:   call void @swift_initializeSuperclass(%swift.type* {{%.*}}, i1 false)
+// CHECK: define{{( protected)?}} private %swift.type* @create_generic_metadata_RootGenericFixedLayout(%swift.type_pattern*, i8**) {{.*}} {
```
Comment (jckarter) [inline test/IRGen/generic_classes.sil]: It shouldn't be possible to be both protected and private.

**Comment by jckarter** [comment]
Looks good.

**Comment by slavapestov** [comment]
I fixed the test, thanks @jckarter. Running tests locally now.

**Comment by gribozavr** [comment]
@swift-ci Please test

**Pull request merged**